tests/rptest/tests/upgrade_test.py

```diff
@@ -91,3 +92,39 @@ def test_basic_upgrade(self):
 
         unique_versions = wait_for_num_versions(self.redpanda, 1)
         assert head_version_str in unique_versions, unique_versions
+
+
+class UpgradeWithWorkloadTest(EndToEndTest):
+    """
+    Test class that performs upgrades while verifying a concurrently running
+    workload is making progress.
+    """
+    def setUp(self):
+        super(UpgradeWithWorkloadTest, self).setUp()
+        # Start at a version that supports rolling restarts.
+        self.initial_version = (22, 1, 3)
+        self.producer_msgs_per_sec = 200
+        install_opts = InstallOptions(version=self.initial_version)
+        self.start_redpanda(num_nodes=3, install_opts=install_opts)
+        self.installer = self.redpanda._installer
+
+        # Start running a workload.
+        spec = TopicSpec(name="topic", partition_count=2, replication_factor=3)
+        self.client().create_topic(spec)
+        self.topic = spec.name
+        self.start_producer(num_nodes=1, throughput=self.producer_msgs_per_sec)
+        self.start_consumer(num_nodes=1)
+        self.await_startup(min_records=self.producer_msgs_per_sec)
+
+    @cluster(num_nodes=5, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_rolling_upgrade(self):
+        self.installer.install(self.redpanda.nodes, RedpandaInstaller.HEAD)
+        # Give ample time to restart, given the running workload.
+        self.redpanda.restart_nodes(self.redpanda.nodes,
+                                    rolling_restart=True,
```

Comment:
i'm not sure, what do you think about `rolling_restart_nodes`?

Reply:
Hmmm I'm indifferent so I'll take it